File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.c
#include <stdio.h>

int maximumDifference(int* nums, int numsSize) {
    int cur = -1;
    for (int i = 0; i < numsSize; ++i) {
        for (int j = i + 1; j < numsSize; ++j) {
            if (nums[j] - nums[i] > cur && nums[j] != nums[i]) {
                cur = nums[j] - nums[i];
            }
        }
    }
    return cur;
}

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        cur=-1
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[j]-nums[i]>cur and nums[j]!=nums[i]:
                    cur=nums[j]-nums[i]
        return cur